sdv/single_table/base.py

```diff
@@ -270,6 +270,44 @@ def get_metadata(self):
         """Return the ``SingleTableMetadata`` for this synthesizer."""
         return self.metadata
 
+    def auto_assign_transformers(self, data):
+        """Automatically assign the required transformers for the given data and constraints.
+
+        This method will automatically set a configuration to the ``rdt.HyperTransformer``
+        with the exact amount of transformers required for it to work.
+
+        Args:
+            data (pandas.DataFrame):
+                The raw data (before any transformations) that will be used to fit the model.
+        """
+        constrained = self._data_processor._fit_transform_constraints(data)
+        columns_created_by_constraints = set(constrained.columns) - set(data.columns)
+        config = self._data_processor._create_config(constrained, columns_created_by_constraints)
+        self._data_processor._create_hyper_transformer_instance()
+        self._data_processor._hyper_transformer.set_config(config)
```

Comment:
here we are missing that the formatters should also be fit. Can we instead make some methods in the `DataProcessor` that do these steps and then call them here?